custom_components/adaptive_lighting/switch.py

```diff
@@ -557,9 +557,7 @@ def __init__(
         self._prefer_rgb_color = data[CONF_PREFER_RGB_COLOR]
         self._separate_turn_on_commands = data[CONF_SEPARATE_TURN_ON_COMMANDS]
         self._take_over_control = data[CONF_TAKE_OVER_CONTROL]
-        self._transition = min(
-            data[CONF_TRANSITION], self._interval.total_seconds() // 2
```

Comment:
Would there be problems now if the transition is larger than the interval?

Reply:
I don't believe so.  For lights that support it, my understanding is that the Transition parameter is sent along with the rest in a single "turn on" call, and then it's up to the light to achieve its own transition.  The only edge case would be if you turn a light off and back on while a transition is occurring then it will come back on without a transition in progress, but then it should be reapplied when Adaptive Lighting intercepts the state change.  If you have a variegated ecosystem with only some bulbs supporting transitions, then you would still want a shorter interval so that the non-transition bulbs can stay in sync -- they would just lose the benefits of the gradual return-to-normal.

That said, I've only tested it with Hue lights, but since the system only ever sends a single "turn on" per interval including the transition, I don't see how it would cause problems on other systems.  What was your original reason for capping the transition to half of the interval?

Reply:
Would you mind setting it back to something more conservative?

```
min(data[CONF_TRANSITION], self._interval.total_seconds())
```

I remember having some weird issues with this, like flickering lights when it was still transitioning and a new update was sent.

Reply:
I would be happy to, but do you think that it would be better to put this
as a warning in the UI if you set the transition to be longer than the
interval?  (Is that even possible?)

I ask because otherwise it's hidden functionality, overriding the user's
setting. It was confusing to me when the transition time wasn't respected
until I saw the code, and I wonder if some of the other tickets about the
transition setting being ignored fall into the same bucket.

Reply:
OK let's just do it as is and see if people complain 😄